packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,544 @@
+                'We strongly recommend backing up your private key. It will be written into the config file, but would you also like to see this sensitive information on screen now?',
+            default: false,
+            transformer(value) {
+                return value ? `Your node's private key: ${privateKey}` : 'no'
+            },
+        })
     }
+
+    return privateKey
 }
 
-const generateApiKey = (): string => {
-    const hex = uuid().split('-').join('')
-    return Buffer.from(hex).toString('base64').replace(/[^0-9a-z]/gi, '')
+/**
+ * Lets the user decide the desired network for their node.
+ */
+async function getNetwork() {
+    return select<'polygon' | 'mumbai'>({
+        message:
+            'Which network do you want to configure your node to connect to?',
+        choices: [
+            { value: 'polygon', name: 'Streamr 1.0 testnet + Polygon' },
+            {
+                value: 'mumbai',
+                name: 'Streamr 1.0 testing environment + Mumbai',
+            },
+        ],
+        default: 'polygon',
+    })
+}
+
+/**
+ * Lets the user gather and configure desired operator plugins.
+ */
+async function getOperatorAddress() {
+    const setupOperator = await confirm({
+        message:
+            'Do you wish to participate in earning rewards by staking on stream Sponsorships?',
+        default: true,
+    })
+
+    if (!setupOperator) {
+        return undefined
+    }
+
+    const operator = await input({
+        message: 'Enter your Operator address:',
+        validate(value) {
+            return isAddress(value) ? true : 'Invalid ethereum address'
+        },
+    })
+
+    return operator.toLowerCase()
 }
 
-export const DEFAULT_CONFIG_PORTS: Record<string, number> = {
-    WS: WebsocketConfigSchema.properties.port.default,
-    MQTT: MqttConfigSchema.properties.port.default,
-    HTTP: BrokerConfigSchema.properties.httpServer.properties.port.default
+interface PubsubPlugin {
+    port?: number
 }
 
-const PLUGIN_NAMES: Record<string, string> = {
-    WS: 'websocket',
-    MQTT: 'mqtt',
-    HTTP: 'http'
+type PubsubPluginKey = 'websocket' | 'mqtt' | 'http'
+
+const DefaultPort: Record<PubsubPluginKey, number> = {
+    websocket: WebsocketConfigSchema.properties.port.default,
+    mqtt: MqttConfigSchema.properties.port.default,
+    http: BrokerConfigSchema.properties.httpServer.properties.port.default,
 }
 
-const PRIVATE_KEY_SOURCE_GENERATE = 'Generate'
-const PRIVATE_KEY_SOURCE_IMPORT = 'Import'
+/**
+ * Lets the user select and configure desired pub/sub plugins.
+ */
+async function getPubsubPlugins() {
+    const setupPubsub = await confirm({
+        message:
+            'Do you wish to use your node for data publishing/subscribing?',
+        default: true,
+    })
 
-export const CONFIG_TEMPLATE: any = {
-    $schema: formSchemaUrl(CURRENT_CONFIGURATION_VERSION),
-    client: {
-        auth: {}
-    },
-    plugins: {},
-    apiAuthentication: {
```

Comment:
Keep the apiAuthentication block. It should contain an auto-generated key if any API plugins (i.e. pubsub plugins) have been enabled.

Reply:
Good to have you, Sir. I'll revert.